astropy/coordinates/tests/test_sky_coord.py

```diff
@@ -1825,3 +1825,28 @@ def test_passing_inconsistent_coordinates_and_units_raises_helpful_error(kwargs,
     # https://github.com/astropy/astropy/issues/10725
     with pytest.raises(ValueError, match=error_message):
         SkyCoord(**kwargs)
+
+
+@pytest.mark.skipif(not HAS_SCIPY, reason="Requires scipy.")
+def test_match_to_catalog_3d_and_sky():
+    # Test for issue #5857. See PR #11449
+    from astropy.coordinates.sky_coordinate import SkyCoord
+    cfk5_default = SkyCoord([4, 2.1] * u.degree, [0, 0] * u.degree, distance=[1, 5] * u.kpc, frame='fk5')
+    cfk5_B1950 = SkyCoord([1, 2, 3, 4] * u.degree, [0, 0, 0, 0] * u.degree, distance=[1, 1, 1, 1.5] * u.kpc,
+                          frame='fk5', equinox='B1950')
+
+    idx, angle, quantity = cfk5_B1950.match_to_catalog_3d(cfk5_default)
+    npt.assert_array_equal(idx, [0, 0, 0, 0])
+    assert_allclose(angle, [2.37584776, 1.38764741, 0.45438139, 0.69810019]*u.deg)
+    assert np.abs(quantity[0].to_value(u.kpc) - 0.04146349) < 1e-6
```

Comment:
I'd write these tests as
```
expected = [0.04146349, 0.02421842, 0.00793043, 0.50022263] * u.kpc
assert_allclose(quantity, expected, atol=1*u.mpc, rtol=0)
```
(and same below)

Reply:
Yep. This is definitely better. When I was writing the tests, I thought of using assert_allclose but wasn't sure how to do so. So, I'll change these that way.